databricks/koalas/indexes.py

```diff
@@ -439,6 +439,39 @@ def values(self):
         warnings.warn("We recommend using `{}.to_numpy()` instead.".format(type(self).__name__))
         return self.to_numpy()
 
+    @property
+    def asi8(self):
+        """
+        Integer representation of the values.
+
+        .. warning:: We recommend using `Index.to_numpy()` instead.
+
+        .. note:: This method should only be used if the resulting NumPy ndarray is expected
+            to be small, as all the data is loaded into the driver's memory.
+
+        Returns
+        -------
+        numpy.ndarray
+            An ndarray with int64 dtype.
+
+        Examples
+        --------
+        >>> ks.Index([1, 2, 3]).asi8
+        array([1, 2, 3])
+
+        Returns None for non-int64 dtype
+
+        >>> ks.Index(['a', 'b', 'c']).asi8 is None
+        True
+        """
+        warnings.warn("We recommend using `{}.to_numpy()` instead.".format(type(self).__name__))
+        if self.dtype == "int64":
```

Comment:
What if the `dtype` is the other integral types? `int8`, `int16`, or `int32`?

```py
>>> pd.Index([1, 2, 3]).asi8
array([1, 2, 3])
>>> pd.Index([1, 2, 3]).astype('int32').asi8
array([1, 2, 3])
>>> pd.Index([1, 2, 3]).astype('int16').asi8
array([1, 2, 3])
>>> pd.Index([1, 2, 3]).astype('int8').asi8
array([1, 2, 3])
```

Reply:
I guess we should use `isinstance(self.spark.data_type, IntegralType)`? (should check the nullability as well?)

Reply:
Thanks fro the comment!
If the null valued included, it should returns `None`.
So I think maybe `isinstance(self.spark.data_type, IntegralType)` is enough!